product_attribute_model_link/models/product_attribute_value.py

```diff
@@ -0,0 +1,185 @@
+# Copyright (C) 2024 Cetmix OÜ
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class ProductAttributeValue(models.Model):
+    _inherit = "product.attribute.value"
+
+    linked_record_ref = fields.Reference(
+        selection="_selection_target_model", string="Linked record"
+    )
+
+    @api.model
+    def _selection_target_model(self):
+        models = self.env["ir.model"].search_read([], ["model", "name"])
```

Comment:
I think need add value in domain `("transient", "=", False)`.

Reply:
You're correct. Fixed!